File: 189-rotate-array/rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        k = k%n
        def reverseArr(start,end):
            while start < end:
                nums[start],nums[end] = nums[end],nums[start]
                start += 1
                end-=1
        #reverse array
     
        reverseArr(0,n-1)

        #reverse  first k element
        reverseArr(0,k-1)

        #reverse remaining element
        reverseArr(k,n-1)        

File: 189-rotate-array/rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) {
        k = k % nums.length;
        reverse(0,nums.length-1,nums);
        reverse(0,k-1,nums);
        reverse(k,nums.length-1,nums);

    }

    public void reverse(int start,int end,int [] arr){
        while(start < end){
            int temp = arr[start];
            arr[start++] = arr[end];
            arr[end--] = temp;
        }
    }
}